src/core/sourcechange.cpp

```diff
@@ -72,5 +72,165 @@ vector<LuaToken> ApplySCVisitor::apply_changes(const vector<LuaToken>& tokens) {
     return new_tokens;
 }
 
+optional<string> default_source_change_label(const val& v) {
+    if (!v.source)
+        return nullopt;
+
+    auto possible_changes = v.forceValue(v);
+
+    if (!possible_changes)
+        return nullopt;
+
+    ApplySCVisitor visitor;
+    (*possible_changes)->accept(visitor);
+    for (const auto& c : visitor.changes) {
+        if (c.hint != "" && c.hint != "?")
+            return c.to_string();
+    }
+
+    return visitor.changes.front().to_string();
+}
+
+vector<string> source_change_labels(const val& v) {
+    if (!v.source)
+        return vector<string>();
+
+    auto possible_changes = v.forceValue(v);
+
+    if (!possible_changes)
+        return vector<string>();
+
+    vector<string> result;
+
+    struct SCLabelVisior : ApplySCVisitor {
+        virtual void visit(const SourceChangeOr& sc_or) override {
+            for (const auto& c : sc_or.alternatives) {
+                c->accept(*this);
+            }
+        }
+
+        virtual void visit(const SourceChangeAnd& sc_and) override {
+            if (!sc_and.changes.empty())
+                sc_and.changes.back()->accept(*this);
+        }
+
+    } visitor;
+
+    (*possible_changes)->accept(visitor);
+    for (const auto& c : visitor.changes) {
+        result.push_back(c.to_string());
+    }
+
+    return result;
+}
+
+/*
+ * removes the first alternative from a source change recursively. It does a
+ * depth first search for a source assignment and removes it, removing empty
+ * or/and nodes in the process.
+ *
+ * Precondition: the source changes must result from a call to x.forceValue(x)
+ * as it compares the match and replacement in the SourceAssignment to discard
+ * any SourceAssignments that do not influence the current source change.
+ * (probably not needed?)
+ *
+ * It returns true if a matching assignment was found, false otherwise.
+ */
+
+bool remove_alternative(shared_ptr<SourceChange>& changes) {
```

Comment:
Helper functions that are not not in a header should be `static` to avoid name clashes when linking.

```suggestion
static bool remove_alternative(shared_ptr<SourceChange>& changes) {
```